### PR TITLE
[AIRFLOW-947] Improve exceptions for unavailable Presto cluster

### PR DESCRIPTION
**Pull request by artwr**
This improves error logging when the Presto cluster is unavailable
and the underlying error is a 503 http response. This introspects
the error to prevent trying to access the 'message' attribute when
not present.

Please accept this PR that addresses the following issues:
- [AIRFLOW-947](https://issues.apache.org/jira/browse/AIRFLOW-947)

Testing Done:
- This would be more of an integration test. I will work on it. Please consider this WIP.

### CODE REVIEW

**Comment by artwr** [comment]
Travis tests except LDAP are passing : https://travis-ci.org/artwr/airflow/jobs/208473660

File: airflow/hooks/presto_hook.py [hunk 1]
```diff
@@ -80,7 +80,14 @@ def get_first(self, hql, parameters=None):
             return super(PrestoHook, self).get_first(
                 self._strip_sql(hql), parameters)
         except DatabaseError as e:
-            raise PrestoException(e[0]['message'])
+            if (hasattr(e, 'message')
```
Comment (mistercrunch) [inline airflow/hooks/presto_hook.py]: Would be nice the add a `self._handle_exception(e)` since the routine is copypasted

**Comment by bolkedebruin** [comment]
@artwr ping

**Comment by artwr** [comment]
@bolkedebruin I addressed Max's comments and refactored. I am not sure what the issue is with apache master, which seems to be not rebasing correctly.

**Comment by bolkedebruin** [comment]
I had some weird issue with git giving this error (on OSX) when pushing a PR:

```
error: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
error: error in sideband demultiplexer
```

I think the sync with GitHub did not kick in.

Your PR seems fine. LGTM.

**Comment by artwr** [comment]
I agree, I think the github sync is just lagging

**Pull request merged**